app/src/main/java/com/github/polypoly/app/menu/JoinGameLobbyActivity.kt

```diff
@@ -491,73 +499,49 @@ class JoinGameLobbyActivity : MenuActivity("Join Game") {
     }
 
     /**
-     * This function is called when the user clicks on the button to join a gameLobby.
+     * This function changes the warning state according to the game lobby code input by the user
      * If the gameLobby code is empty, or the code is not in the DB, or the gameLobby is full,
      * it displays a warning message.
      * Otherwise, it calls the function to join the gameLobby.
      * @param warningState (MutableState<String>): The state of the warning message
      * @param mContext (Context): The context of the activity
-     * @return (String): The warning message to be displayed
      */
+    @SuppressLint("NewApi")
     private fun gameLobbyCodeButtonOnClick(warningState: MutableState<String>, mContext: Context) {
-        return if (gameLobbyCode.isEmpty()) {
+        if (gameLobbyCode.isEmpty()) {
             warningState.value = getString(R.string.game_lobby_code_is_empty)
-        } else if (!dbContainsGameLobbyCode(gameLobbyCode)) {
-            warningState.value = getString(R.string.game_lobby_does_not_exist)
-        } else if(gameLobbyIsFull(gameLobbyCode)){
-            warningState.value = getString(R.string.game_lobby_is_full)
         } else {
-            warningState.value = ""
-            joinGameLobbyRoom(mContext)
+            val lobbyKey = DB_GAME_LOBIES_PATH + gameLobbyCode
+            remoteDB.keyExists(lobbyKey).thenCompose { keyExists ->
+                if (!keyExists) {
+                    warningState.value = getString(R.string.game_lobby_does_not_exist)
+                    CompletableFuture.failedFuture(IllegalAccessException())
```

Comment:
this is not a great choice of exception, `IllegalAccessException` is intended for reflection operations, perhaps `NoSuchElementException` instead?